src/map/skill.cpp

```diff
@@ -8839,12 +8900,19 @@ int skill_castend_nodamage_id (struct block_list *src, struct block_list *bl, ui
 #endif
 	case PR_MAGNIFICAT:
 	case PR_GLORIA:
+	case SOA_SOUL_OF_HEAVEN_AND_EARTH:
 		if (sd == NULL || sd->status.party_id == 0 || (flag & 1)) {
 
 			// Animations don't play when outside visible range
 			if (check_distance_bl(src, bl, AREA_SIZE))
 				clif_skill_nodamage(bl, bl, skill_id, skill_lv, 1);
 
+			if( skill_id == SOA_SOUL_OF_HEAVEN_AND_EARTH ){
```

Comment:
```suggestion
			if( skill_id == SOA_SOUL_OF_HEAVEN_AND_EARTH ){
				status_percent_heal( bl, 0, 100 );

```
Missing SP healing of target.